apps/sms/views/shared/js/settings.js

```diff
@@ -72,6 +79,15 @@ var Settings = {
     }
   },
 
+  // return resolve promise with sendReadReport status
+  initSendReadReport: function initSendReadReport(value) {
+    this._sendReadReportDefer.resolve(value);
+    delete this._sendReadReportDefer;
+
+    navigator.mozSettings.addObserver('messages.mms.sendReadReport.enabled',
+      (e) => this.sendReadReport = Promise.resolve(e.settingValue));
```

Comment:
If we apply settings addObserver init pattern like MaxConcatenatedMsg and others below, why do we need to return as promise here instead of return the value directly?